src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,166 @@
+
+    def __init__(self, value: float):
+        super().__init__(value)
+
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the ClosedBound as the lower Bound of an interval."""
+        return f"[{self}"
+
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the ClosedBound as the upper Bound of an interval."""
+        return f"{self}]"
+
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value is not strictly lower than the ClosedBound."""
+        return value >= self._value
+
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value is not strictly higher than the ClosedBound."""
+        return value <= self._value
+
+
+class OpenBound(Bound):
+    """
+    An open Bound, i.e. the value on the border does not belong to the range.
+
+    May be infinite (unbounded).
+    """
+
+    def __init__(self, value: float):
+        super().__init__(value)
+
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the OpenBound as the lower Bound of an interval."""
+        return f"({self}"
+
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the OpenBound as the upper Bound of an interval."""
+        return f"{self})"
+
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value is not lower or equal to the Bound."""
+        return value > self._value
+
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value is not higher ot equal to the Bound."""
+        return value < self._value
+
+
+class _Infinity(OpenBound):
+    """An infinite or unrestricted upper Bound."""
+
+    def __init__(self) -> None:
+        super().__init__(float("inf"))
+
+    def __str__(self) -> str:
+        """Get a string representation of the concrete value of the Bound."""
+        return "\u221e"
+
+    def check_lower_bound(self, _: float) -> bool:
+        """Check that a value does not exceed the unrestricted Bound on the lower side. Always False."""
+        return False
+
+    def check_upper_bound(self, _: float) -> bool:
+        """Check that a value does not exceed the unrestricted Bound on the upper side. Always true."""
+        return True
+
+
+class _MinInfinity(OpenBound):
+    """An infinite or unrestricted lower Bound."""
+
+    def __init__(self) -> None:
+        super().__init__(float("-inf"))
+
+    def __str__(self) -> str:
+        """Get a string representation of the concrete value of the Bound."""
+        return "-\u221e"
+
+    def check_lower_bound(self, _: float) -> bool:
+        """Check that a value does not exceed the unrestricted Bound on the lower side. Always true."""
+        return True
+
+    def check_upper_bound(self, _: float) -> bool:
+        """Check that a value does not exceed the unrestricted Bound on the upper side. Always false."""
+        return False
```

Comment:
I'm still not quite happy with these classes:

* `_Infinity` inherits from `OpenBound`. The latter can be used as upper and as lower bound, the former only as an upper bound. Thus, `_Infinity` cannot be used everywhere an `OpenBound` can be used, breaking Liskov's Substitution Principle.
* Similar argument for `_MinInfinity`

I'd rather remove them and adjust the `__str__` methods of `OpenBound` and `ClosedBound` to handle the cases `float("inf")` and `float("-inf")` as you're doing here.

Reply:
Makes sense, did that now.